feat(dashboard/invoices): add ability to create invoices

CodeNamedRobin opened the pull request:
# Description
Adds invoice user table showing invoice users with outstanding debts, and allows for creating invoices. immediately.

## Related issues/external references
## Types of changes
<!-- What types of changes does your code introduce? Remove all the items that do not apply: -->
- New feature _(non-breaking change which adds functionality)_

CodeNamedRobin committed:
feat(dashboard/invoices): added invoice users table

CodeNamedRobin committed:
feat(dashboard/invoice): added create invoice modal

feat(dashboard/forms): added switchtoggle inputspan

CodeNamedRobin committed:
feat(dashboard/invoice): made create invoice modal route to invoice view after completion

CodeNamedRobin committed:
feat(dashboard/invoice): added localisation to invoiceusertable and invoicecreatemodal

CodeNamedRobin committed:
refactor(dashboard/invoice): fix type-check errors

SuperVK commented:
![image](https://github.com/user-attachments/assets/5243fd36-f4d2-4c70-bb75-0742d45dae2e)
Maybe I'm missing something, but this does not seem to work?

CodeNamedRobin committed:
feat(dashboard/invoices): added description to invoice user table

feat(dashboard/invoices): added string to invoice user table if there are no users with outstanding debts

CodeNamedRobin commented:
> ![image](https://private-user-images.githubusercontent.com/31625323/358219802-5243fd36-f4d2-4c70-bb75-0742d45dae2e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM5MTYyNzcsIm5iZiI6MTcyMzkxNTk3NywicGF0aCI6Ii8zMTYyNTMyMy8zNTgyMTk4MDItNTI0M2ZkMzYtZjRkMi00YzcwLWJiNzUtMDc0MmQ0NWRhZTJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE3VDE3MzI1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU0NjM0MjNiMTk2ZDkxMDZjYjExOGUzNTczYTVlZjE5ZTFjYWI1MzhjM2Q3ZGY5MjMwZDdmYTlhMTYyNTU2NjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.x2QoVb6GVKVmFrlaVftEjwPTxc0ftDRdXEcE1ypahxQ) Maybe I'm missing something, but this does not seem to work?

Hi Victor,
It should work, but is dependent on if invoice users have outstanding debts. In the latest commit I have added a string if there are no outstanding debts by invoice users. Should clear the confusion.

SuperVK commented:
I cannot see the string, there are translations missing as well. And I can't seem to link transactions to the invoice, or add some sort of amount to the invoice? 

Again, maybe I'm missing something here though. But then the UX might need to be a bit clearer?

CodeNamedRobin commented:
> I cannot see the string, there are translations missing as well. And I can't seem to link transactions to the invoice, or add some sort of amount to the invoice?
> 
> Again, maybe I'm missing something here though. But then the UX might need to be a bit clearer?

Woops, I did not push the commit, should work now.

SuperVK commented on apps/dashboard/src/utils/validation-schema.ts:
```diff
@@ -99,4 +99,18 @@ export const addContainerObject =
+      postalCode: yup.string().required(),
+      city: yup.string().required(),
+      country: yup.string().required(),
+      attention: yup.string().required(),
```
Attention is not required by the backend

CodeNamedRobin replied:
Fixed in latest commit.

CodeNamedRobin committed:
fix(dashboard/invoices): fixed attention being required by the frontend in create invoice form

CodeNamedRobin committed:
fix(dashboard/invoices): type check issues on invoice craete form

SuperVK commented:
The translations are still missing. 

I also at this point would advise against merging this with the state of the backend. We can't test the code because of the 501 it's throwing, and we expect some spec changes as well. It will just be broken code that needs to be rewritten either way. If you really want I won't stop you and we can still merge, but I think it will just be a very confusing lifeless components that are not much useful in this state.

SuperVK commented:
Closed on request of Robin
